Recordings in paths

rcmarron opened the pull request:
## Changes

Adds recordings to the persons endpoint in paths. Part of (#7246)

Notes:
* It's behind the feature_flag `recordings-in-insights`. There should be no changes to the query if this feature flag is not set.
* It made sense refactor the `include_recordings` flag from the trends person modal into a filter because of some multi-inheritance issues, so that's included this PR


## How did you test this code?

Added unit tests to the query + made sure no changes occurred on snapshots of previous tests.

rcmarron committed:
add recordings to path query

rcmarron committed:
Merge branch master of https://github.com/PostHog/posthog into recordings-in-paths

rcmarron committed:
uncomment cache

rcmarron committed:
add clarifying comment

rcmarron committed:
works for start/end paths

rcmarron committed:
move to extra fields/properties

rcmarron committed:
add tests

rcmarron committed:
cleanup

rcmarron committed:
update ff name

rcmarron committed:
Merge branch master of https://github.com/PostHog/posthog into recordings-in-paths

rcmarron committed:
fix flaky test

rcmarron committed:
Merge branch master of https://github.com/PostHog/posthog into recordings-in-paths

EDsCODE commented on ee/clickhouse/queries/test/test_paths.py:
```diff
@@ -2758,6 +2780,215 @@ def test_path_groups_filtering(self):
+                [
+                    {
+                        "session_id": "s1",
+                        "events": [
```
Is this only supposed to have 1 event? What about the start event

rcmarron replied:
Right now, it only includes the event of the step in the path that the user clicked on.

I could definitely see us wanting to expand this to include all of the events from the path, but as it stands today, paths can span recordings, so we can't reliably support that feature. And it would be a bit of work to get paths be contained to a recording's session. My vote is we get this feature out, and wait for people to ask for the feature first.

EDsCODE reviewed (commented):
Overall looks good. Had a question and might make sense to add some extra cases around path dropoff sessions

rcmarron committed:
test and handle path_dropoff_key case

rcmarron committed:
Merge branch master of https://github.com/PostHog/posthog into recordings-in-paths

rcmarron commented:
Good call on the `path_dropoff_key` case. There was a bug where it was returning the wrong event in that situation. Fixed the bug and added a test for it.
